pkg/analyzer/nodes/node.go

```diff
@@ -56,7 +56,7 @@ type Components struct {
 }
 
 // Analyze performs analysis of node's hosts filesystem and return the detected components.
-func Analyze(nodeName, rootFSdir string, uncertifiedRHEL bool) (*Components, error) {
+func Analyze(nodeName, rootFSdir string, opts detection.DetectComponentOpts) (*Components, error) {
```

Comment:
not sure if this is overkill, but I think technically we should probably have a separate set of options for `Analysis` which currently, coincidentally matches the options in `detection.DetectComponentOpts`

Reply:
This is okay, but I personally would prefer having a separate options type (i.e. not reuse `detection.DetectComponentOpts`) as Ross suggested.